adapter/service/ota/service.go

```diff
@@ -0,0 +1,161 @@
+package ota
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/futurehomeno/fimpgo"
+	"github.com/futurehomeno/fimpgo/fimptype"
+
+	"github.com/futurehomeno/cliffhanger/adapter"
+)
+
+// Controller is an interface representing a device capable of OTA firmware updated.
+// In a polling scenario implementation might require some safeguards against excessive polling.
+type Controller interface {
```

Comment:
If we need to we might split this controller in future into two, one for starting, which is optional, and one for reporting which is mandatory. Zigbee for instance does its OTA on its own and does not allow external trigger. But it's just a remark for future.